docs/src/main/sphinx/sql/match-recognize.rst

```diff
@@ -612,6 +612,112 @@ This is incorrect::
 
     LAST(A.totalprice + B.totalprice)
 
+Aggregate functions
+^^^^^^^^^^^^^^^^^^^
+
+It is allowed to use aggregate functions in a row pattern recognition context.
+Aggregate functions are evaluated over all rows of the current match or over a
+subset of rows based on the matched pattern variables. The
+:ref:`running and final semantics<running_and_final>` are supported, with
+``running`` as the default.
+
+The following expression returns the average value of the ``totalprice`` column
+for all rows matched to pattern variable ``A``::
+
+    avg(A.totalprice)
+
+The following expression returns the average value of the ``totalprice`` column
+for all rows matched to pattern variables from subset ``U``::
+
+    avg(U.totalprice)
+
+The following expression returns the average value of the ``totalprice`` column
+for all rows of the match::
+
+    avg(totalprice)
+
+Aggregation arguments
+"""""""""""""""""""""
+
+In case when the aggregate function has multiple arguments, it is required that
+all arguments refer consistently to the same set of rows::
+
+    max_by(totalprice, tax) /* aggregate over all rows of the match */
+
+    max_by(CLASSIFIER(A), A.tax) /* aggregate over all rows matched to A */
+
+This is incorrect::
+
+    max_by(A.totalprice, tax)
+
+    max_by(A.totalprice, A.tax + B.tax)
+
+If an aggregate argument does not contain any column reference or
+``classifier`` function, it does not refer to any pattern variable. In such a
+case the universal row pattern variable is implicit for the first argument.
+This means that the aggregate function applies to all rows of the match::
+
+    count(1) /* aggregate over all rows of the match */
+
+Each of the subsequent arguments must either refer to the pattern variable
+established by the first argument, or not refer to any pattern variable::
+
+    min_by(1, 2) /* aggregate over all rows of the match */
+
+    min_by(1, totalprice) /* aggregate over all rows of the match */
+
+    min_by(totalprice, 1) /* aggregate over all rows of the match */
+
+    min_by(A.totalprice, 1) /* aggregate over all rows matched to A */
+
+This is incorrect::
+
+    max_by(1, A.totalprice)
```

Comment:
The spec does not discuss the case of multiple aggregate arguments.
Considering arguments that do not refer to any label (because they have no column references or `classifier()` calls) I think that they should "inherit" the label from other arguments. I arbitrarily decided that we require the first argument to establish the label, but we could also allow the other way around.

Please check if you can find the missing information in the spec, and let us decide afterwards how to handle this case.

Reply:
The spec says:

> If `AF` is contained in a `<row pattern measure expression>` or a `<row pattern definition search condition>`, and **if an aggregated argument of `AF` contains a column reference or a `<classifier function>`, then there shall be exactly one row pattern variable `RPV` such that `RPV` qualifies every column reference contained in an aggregated argument of `AF`** and the explicit or implicit argument of every `<classifier function>` contained in an aggregated argument of `AF` is `RPV`.

From that, I interpret that if any column is qualified with a label, or if there's any `classifier` call, then *all* columns and `classifier()` calls need to be qualified with the same label.

Therefore, `max_by(1, A.totalprice)` and `max_by(A.totalprice, 1)` should be treated the same way. They should both be valid because there's a single column reference.

On the other hand, `f(a, X.b)` and `f(X.a, b)` are both invalid because one column is labeled with `X` but the other is not.

Reply:
Actually, I saw that piece, but I think that it only discusses dependencies within a single aggregated argument.
Anyway, I made the suggested change, since it looks reasonable.